### ISSUE TITLE
Installing fomantic-ui with yarn - gulp install fails

### ISSUE THREAD

**Issue by JamesYFC** [fomantic/Fomantic-UI#811]
# Help Wanted

## Problem
Installation with yarn is not working when running with the following

`yarn add fomantic-ui --ignore-scripts`
`yarn --cwd node_modules/fomantic-ui run install`

the install (autoInstall mode) gulp task seems to fail due to missing semantic.json?

logs: 
```
yarn run v1.16.0
$ gulp install
[11:58:09] Using gulpfile C:\Projects\TESTTelemetry\create-react-app\rsg-telemetry\front\node_modules\fomantic-ui\gulpfile.js
[11:58:09] Starting 'install'...
[11:58:09] Starting 'run setup'...
[11:58:09] Finished 'run setup' after 1 ms
[11:58:09] Starting 'create install files'...
Auto-Installing (Without User Interaction)
------------------------------
[11:58:09] 'create install files' errored after 1.63 ms
[11:58:09] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:427:7)
    at C:\Projects\TESTTelemetry\create-react-app\rsg-telemetry\front\node_modules\fomantic-ui\tasks\install.js:273:28
    at taskWrapper (C:\Projects\TESTTelemetry\create-react-app\rsg-telemetry\front\node_modules\undertaker\lib\set-task.js:13:15)
    at bound (domain.js:402:14)
    at runBound (domain.js:415:12)
    at asyncRunner (C:\Projects\TESTTelemetry\create-react-app\rsg-telemetry\front\node_modules\async-done\index.js:55:18)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[11:58:09] 'install' errored after 6.38 ms
error Command failed with exit code 1.
```

more details below...

https://github.com/fomantic/Fomantic-UI/issues/386#issuecomment-501092373
https://github.com/fomantic/Fomantic-UI/issues/386#issuecomment-501099409

**Comment by robclancy** [comment]
I came to this package instead of semantic hoping at the very least they had installing working. But here we are still... 2020 and a css framework won't install properly because it hacks in its own installation crap.

**Comment by wilpar** [comment]
i'm on a quest to have a fully functional install on rails 6, without legacy app/assets.  A yarn install would 'complete me'.

**Comment by kokogakayui** [comment]
Same problem here with a simple yarn add fomantic-ui. That would also help me go full webpack on Rails 6...

**Comment by cimm** [comment]
For Rails there is the [Fomantic-UI-SASS gem ](https://github.com/fomantic/Fomantic-UI-SASS) which seems to work (only tried the styling, not tried the JavaScript) but a few patch levels behind.

**Comment by davidroeca** [comment]
The commands initially described in the issue actually work for me, though I'm using yarn workspaces, so it works, but a little differently:

```
yarn add fomantic-ui --ignore-scripts
yarn --cwd ../../node_modules/fomantic-ui run install
```

However, now I need to define a `.yarnrc` file with the following content:

```
# Wait for https://github.com/fomantic/Fomantic-UI/issues/811 to be solved
--ignore-scripts true
```

That will lead to some weird issues if another package I'm relying uses these sorts of scripts, but it's what I'm stuck with for now.

**Comment by davidroeca** [comment]
Hit another snag. When I handle the steps of the post-install script, I have to manually edit the project root since it's not the repository root. Not a great first installation experience, but fine, it only happens once.

However, when someone clones my repo and tries to run `yarn install --frozen-lockfile`, the postinstall script tries to run again. Reproduced the issue [here](https://github.com/davidroeca/reproduce-fomantic-ui-811).

**Comment by davidroeca** [comment]
Is there any way this script could not be invoked by default but rather be run via a separate CLI? It seems like something that’s only useful when creating a new project, not something that needs to be run every time. So a helper CLI would probably be preferable e.g. a script that could be run via `npx fomantic-setup new-project-dir` or something to that effect. Maybe an upgrade utility needs to be added as well.

Another option could be to allow an [environment variable of some sort to disable this script](https://stackoverflow.com/a/50402065). I like this option less, since that environment variable will need to be set on every install for my project, but it would probably lead to fewer breaking changes and is probably easier to implement.

**Comment by iamareebjamal** [comment]
Hi, couldn't contribute here due to know indication about what the possible course of action should be? SUI's last usable version was installable through yarn as it is currently used by us in the project. This happened only with fomantic UI. I don't thus know what's the stance with the current package, etc but if the customization prompt is to be kept, then it should be separated from the core package and depend on it instead so that people can install core package without any prompt who don't want any customization. For example @fomantic-ui/core can be core package and fomantic-ui can depend on it and prompt users. This way, we'll just use @fomantic-ui/core and be done with it.

**Comment by davidroeca** [comment]
Commenting here again to mention another work-around I've found by using yarn 2 (berry): https://yarnpkg.com/configuration/manifest#dependenciesMeta

```js
// package.json
{
  // ...
  "dependenciesMeta": {
    "fomantic-ui": {
      "enableScripts": false
    }
  }
}
```

So I'd recommend also trying to upgrade to yarn 2 to resolve this particular issue.

**Comment by robclancy** [comment]
> I came to this package instead of semantic hoping at the very least they had installing working. But here we are still... 2020 and a css framework won't install properly because it hacks in its own installation crap.
2021

**Comment by lubber-de** [comment]
This issue is tagged as "help-wanted", so you are welcome to help out and provide a PR to fix/support it 😉

**Comment by lubber-de** [comment]
Should now be fixed in #2272 because that wont start the interactive script automatically anymore
